docs/sql-migration-guide.md

```diff
@@ -97,7 +97,7 @@ license: |
 
   - Since Spark 3.0, when Avro files are written with user provided non-nullable schema, even the catalyst schema is nullable, Spark is still able to write the files. However, Spark will throw runtime NPE if any of the records contains null.
 
-  - Since Spark 3.0, we use a new protocol for fetching shuffle blocks, for external shuffle service users, we need to upgrade the server correspondingly. Otherwise, we'll get the error message `UnsupportedOperationException: Unexpected message: FetchShuffleBlocks`. If it is hard to upgrade the shuffle service right now, you can still use the old protocol by setting `spark.shuffle.useOldFetchProtocol` to `true`.
+  - Since Spark 3.0, we use a new protocol for fetching shuffle blocks, for external shuffle service users, we need to upgrade the server correspondingly. Otherwise, we'll get the error message `IllegalArgumentException: Unexpected message type: <number>`. If it is hard to upgrade the shuffle service right now, you can still use the old protocol by setting `spark.shuffle.useOldFetchProtocol` to `true`.
```

Comment:
Since you're changing this, this item is pretty hard to read. Better:

```
Since Spark 3.0, a new protocol for fetching shuffle blocks is used. It's recommended that external shuffle services
be upgraded when running Spark 3.0 apps. Old external shuffle services can still be used by setting the configuration
`spark.shuffle.useOldFetchProtocol` to `true`. Otherwise, Spark may run into errors with messages like `Illegal ...`.
```

(On a side note, the "Since Spark blah.blah" thing on every item is pretty redundant; having proper subsections would be better, like in the core guide. But not something to change here. But now that I mention this, this item really should be in the core guide, not in the SQL guide.)

Reply:
I have moved it to core migration guide and removed the "Since Spark 3.0, " as having subsections with the Spark versions.